src/site/content/en/accessible/labels-and-text-alternatives/index.md

```diff
@@ -334,3 +334,105 @@ promotional offers?" like in the VoiceOver example below:
   alt="VoiceOver text output showing 'Receive promotional offers?'">
 </figure>
+
+{% AssessmentCallout 'Use the drop-down below each code sample to check whether the HTML element in the sample has an accessible name.' %}
+{% Tabs 'Samples for knowledge self check' %}
+{% Tab 'sample' %}
+
+```html
+<img src="temp.svg" alt="temp">
+```
+
+{% AssessmentHint 'Does this image have an accessible name?' %}
+**Yes**. Give an image an accessible name by adding an `alt` attribute.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+```html
+<object type="application/pdf" data="/report.pdf" alt="Annual report"></object>
+```
+
+{% AssessmentHint 'Does this object have an accessible name?' %}
+**No**. Give an `<object>` element a name by providing text content.
+(An `alt` attribute on an object will not be read by assistive technologies.)
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+This sample renders an icon button:
+
+```html
+<button>
+  <svg class="icon">
+    <use xlink:href="#icon-1" />
+  </svg>
+</button>
+```
+
+{% AssessmentHint 'Does this button have an accessible name?' %}
+**No**. Give a button an accessible name by providing text content,
+an `aria-label` attribute, or an `aria-labelledby` attribute.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+This sample renders an icon button with a tooltip that appears on hover and focus:
+
+```html
+<button>
+  <svg class="icon">
+    <use xlink:href="#icon-1" />
+  </svg>
+  <span class="tooltip hidden">Open menu</span>
+</button>
+```
+
+{% AssessmentHint 'Does this button have an accessible name?' %}
+**Yes**. Even if the button's tooltip is visually hidden using CSS,
+assistive technologies can still read its text content.
+
+**Caution:** When visually hiding button text,
+use CSS rather than the `hidden` attribute.
+The `hidden` attribute removes the element from the accessibility tree.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+```html
+<label>
+  <input type="checkbox" value="subscribe"></input>
+  Subscribe to the newsletter?
+</label>
+```
+
+{% AssessmentHint 'Does this checkbox have an accessible name?' %}
+**Yes**. The label is associated with the checkbox
+because the label is the checkbox's parent.
+So, assistive technologies treat the label's text content as the input's name.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% Tab 'sample' %}
+
+```html
+<input type="checkbox" value="subscribe"></input>
+<label for="subscribe">Subscribe to the newsletter?</label>
+```
+
+{% AssessmentHint 'Does this checkbox have an accessible name?' %}
+**No**. The checkbox's `value` attribute indicates
+what text will be saved when the form is submitted,
+but assistive technologies don't read it.
+To provide an accessible name,
+add an `id` attribute with the value `subscribe`
+to associate the checkbox with the label.
+{% endAssessmentHint %}
+
+{% endTab %}
+{% endTabs %}
+{% endAssessmentCallout %}
```

Comment:
Got a couple of these wrong and learned some stuff!

I'll note that I didn't read the page's content and instead jumped straight to the self-assessment. Which honestly doesn't seem bad IMO. It leads me to an interesting idea. We could lead off articles with "how much do you know?" self-assessments, and the answers would have links that jump users to the explanations for the things they got wrong... pretty efficient way to learn... you skip over the stuff you already know, and focus on the stuff you get wrong

cc @mfriesenhahn @robdodson @devnook

Reply:
So, one note: once we start cranking out more sets, I do think reviewers should read the parent post first. If we ask questions about concepts not covered in the post, they're gonna seem unfair and are likely to lead to negative reactions. ("How was I supposed to know that? You didn't tell me!")

That said, what you're talking about is typically called pre-assessment or formative assessment. I'm a little gun shy about communicating to users that they don't need to read an article if they answer the questions correctly. To have high confidence that a user has achieved mastery of a domain of any complexity, you usually need them to answer a fair number of questions that you're pretty confident are all accurately measuring that domain. All of that is a non-trivial amount of work in the short term, and I don't want to make promises to our users that may not be true. However, in the long term, I think it'd be awesome to experiment with a larger item bank that can be organized into quizzes presented as peers to articles [the way Khan Academy does](https://www.khanacademy.org/humanities/grammar/parts-of-speech-the-noun).